renderdoc/os/win32/win32_shellext.cpp

```diff
@@ -487,9 +499,33 @@ struct RDCThumbnailProvider : public IThumbnailProvider, IInitializeWithStream
               case ResourceFormatType::BC1:
               case ResourceFormatType::BC2:
               case ResourceFormatType::BC3:
-                memcpy(decompImagePointer, decompPointer,
-                       16);    // copy one stride of the decompressed Block
-                decompPointer += 16;
+              case ResourceFormatType::BC7:
+                if(resourceType == ResourceFormatType::BC7 && isSRGB)
+                {
+                  for(int pixelInStride = 0; pixelInStride < 4; pixelInStride++)
+                  {
+                    int pixelIndex = pixelInStride * 4 + (i * 16);
+                    // get srgb value as float
+                    Vec4f srgb = Vec4f(decompPointer[pixelIndex] / 255.0f,
+                                       decompPointer[pixelIndex + 1] / 255.0f,
+                                       decompPointer[pixelIndex + 2] / 255.0f, 0.0f);
+                    Vec4f rgb = ConvertSRGBToLinear(srgb);
```

Comment:
Why are we doing an sRGB decode here? I assume that the output thumbnail is expected to be in sRGB space, and we're not decoding any of BC1-BC3 which are likely to be also stored in sRGB. Arguably we should check to see if the dds file is using the linear format and do a conversion _to_ sRGB, but as with the texture viewer I think it's more conventional to simply let explorer interpret the linear data as sRGB without conversion (as most viewers will do that).

Reply:
Removed the conversion from srgb to linear.